client-spark/common/src/main/java/org/apache/spark/shuffle/reader/RssShuffleDataIterator.java

```diff
@@ -117,17 +117,20 @@ public boolean hasNext() {
       // If ShuffleServer delete
 
       ByteBuffer rawData = rawBlock != null ? rawBlock.getByteBuffer() : null;
-      long fetchDuration = System.currentTimeMillis() - startFetch;
-      shuffleReadMetrics.incFetchWaitTime(fetchDuration);
+      long readDuration = System.currentTimeMillis() - startFetch;
       if (rawData != null) {
         uncompress(rawBlock, rawData);
         // create new iterator for shuffle data
         long startSerialization = System.currentTimeMillis();
         recordsIterator = createKVIterator(uncompressedData);
         long serializationDuration = System.currentTimeMillis() - startSerialization;
-        readTime += fetchDuration;
+        long fetchDuration = System.currentTimeMillis() - startFetch;
+        shuffleReadMetrics.incFetchWaitTime(fetchDuration);
+        readTime += readDuration;
         serializeTime += serializationDuration;
```

Comment:
I think we just need to add the ser + decompression into the metric like this.

shuffleReadMetrics.incFetchWaitTime(serializationDuration + uncompression);

Reply:
Hi @zuston 

Thank you for getting back to me so quickly and for reminding me. I have updated the part you mentioned.
Our new implementation does not seem to fit the original test case because it takes the `fetchDuration`(network transmission) as an assumption. Do you think we should change the test case here?

https://github.com/apache/uniffle/blob/1e48bc673d1c0ee41f889a0de6192b0fab131467/client-spark/common/src/test/java/org/apache/spark/shuffle/reader/RssShuffleDataIteratorTest.java#L337

Reply:
Hi @zuston 

Thanks and I updated the part you mentioned. Any suggestions would be appreciated if you are available. 😄